### PR TITLE
Reduce number of CI jobs run on PRs

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

**Review by DMRobertson — DISMISSED** [review]
SGTM, but we should confirm on develop that we run all the jobs we intend to. (I'd be okay with just running against 3.7 and 3.10 on develop FWIW).

Other ideas for reducing runner contention on PRs:
- Only try to build debian packages on PRs if there are changes to the `debian` directory, `build_debian_packages`?
- Run fewer sytest combinations
- Don't test portdb against 3.10/14
- Merge some of the linting steps into one job

**Comment by erikjohnston** [comment]
> SGTM, but we should confirm on develop that we run all the jobs we intend to. (I'd be okay with just running against 3.7 and 3.10 on develop FWIW).
> 
> Other ideas for reducing runner contention on PRs:
> 
> * Only try to build debian packages on PRs if there are changes to the `debian` directory, `build_debian_packages`?
> * Run fewer sytest combinations
> * Don't test portdb against 3.10/14
> * Merge some of the linting steps into one job

Yup good points, though for the linting/portdb jobs they tend to go fast enough that I don't think its a huge deal. Its things that take 20+mins that are the main issue I think.

File: .ci/scripts/calculate_jobs.py [hunk 1]
```diff
@@ -58,17 +62,67 @@
...
+    {
+        "sytest-tag": "focal",
+    },
+    {
+        "sytest-tag": "focal",
+        "postgres": "postgres",
+    },
+    {
+        "sytest-tag": "focal",
+        "postgres": "mulit-postgres",
```
Comment (DMRobertson) [inline .ci/scripts/calculate_jobs.py:102]: typo!

Comment (DMRobertson) [reply .ci/scripts/calculate_jobs.py:102]: also on 116

File: .ci/scripts/calculate_jobs.py [hunk 2]
```diff
@@ -58,17 +62,67 @@
...
+
+
+print("::group::Calculated sytest jobs")
+print(json.dumps(sytest_tests, indent=4))
 print("::endgroup::")
 
-test_matrix = json.dumps(sqlite_tests + postgres_tests + no_extra_tests)
-print(f"::set-output name=test_matrix::{test_matrix}")
+test_matrix = json.dumps(sytest_tests)
+print(f"::set-output name=sytest_test_matrix::{test_matrix}")
```
Comment (DMRobertson) [inline .ci/scripts/calculate_jobs.py:128]: I think you still want to output the trial test matrix?

Comment (erikjohnston) [reply .ci/scripts/calculate_jobs.py]: That is still happening above

**Review by DMRobertson — APPROVED** [review]
LGTM, but please check that this works as intended on develop!

**Comment by erikjohnston** [comment]
> LGTM, but please check that this works as intended on develop!

https://github.com/matrix-org/synapse/actions/runs/2993220092 seems to have the expected number of jobs :tada:

**Pull request merged**